packages/sync/src/VoyagerClient.ts

```diff
@@ -0,0 +1,28 @@
+import ApolloClient, { OperationVariables, MutationOptions, ApolloClientOptions } from "apollo-client";
+import { NormalizedCacheObject } from "apollo-cache-inmemory";
+import { FetchResult } from "apollo-link";
+import ProxyUpdate from "./ProxyUpdate";
+import { OperationDefinitionNode } from "graphql";
+
+export class VoyagerClient extends ApolloClient<NormalizedCacheObject> {
+  private readonly proxyUpdate?: ProxyUpdate;
+
+  constructor(options: ApolloClientOptions<NormalizedCacheObject>, proxyUpdate?: ProxyUpdate) {
+    super(options);
+
+    this.proxyUpdate = proxyUpdate;
+  }
+
+  public mutate<T, TVariables = OperationVariables>(options: MutationOptions<T, TVariables>): Promise<FetchResult<T>> {
```

Comment:
We could chat about this. Why we need to wrap mutate. I was hopping to go different way and make sync really lightweight thing and have set of separate packages that extends apollo client. Is there any limitation in actual client?

Reply:
yeah, probably better to do this differently. I went with approach from the article you linked in your other comment. This way it is not necessary to specify update functions again for each mutation. Last commit removes the wrapper.

Reply:
Oh.. no.. Let's not made changes until we will have idea how this works.